src/VisionModel/resnet.jl

```diff
@@ -0,0 +1,78 @@
+using Flux
+
+basicblock(inplanes::Int, outplanes::Int, downsample::Bool = false) = downsample ? 
+  Chain(Conv((3, 3), inplanes => outplanes, stride = 1, pad = 1),
+        BatchNorm(outplanes, λ = relu),
+        Conv((3, 3), outplanes => outplanes, stride = 2, pad = 1,
+        BatchNorm(outplanes, λ=relu)) : 
+  Chain(Conv((3, 3), inplanes => outplanes, stride = 1, pad = 1),
+        BatchNorm(outplanes, λ = relu),
+        Conv((3, 3), outplanes => outplanes, stride = 1, pad = 1,
+        BatchNorm(outplanes, λ=relu))
+
+bottleneck(inplanes::Int, outplanes::Int, downsample::Bool = false) = downsample ?
+  Chain(Conv((1, 1), inplanes => inplanes, stride = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => inplanes, stride = 1, pad = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((1, 1), inplanes => outplanes, stride = 2),
+        BatchNorm(outplanes, λ = relu)) :
+  Chain(Conv((1, 1), inplanes => inplanes, stride = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => inplanes, stride = 1, pad = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((1, 1), inplanes => outplanes, stride = 1),
+        BatchNorm(outplanes, λ = relu))
+
+projection(inplanes::Int, planes::Int, stride::Int) = Chain(Conv((1, 1), inplanes => outplanes, stride=stride),
+                                                            BatchNorm((outplanes), λ=relu))
+
+
+function resnet(channel_config::Dict{string, Int}, block_config::Dict{String, Int})
+  #=  begin
+    planes = 64
+    inplanes = 64
+    expansion = 1
+    stride = 1
+    end =#
+    layers = []
+    push!(layers, Conv((7, 7), 3=>inplanes, stride=(2, 2), pad=(3, 3)))
+    push!(layers, BatchNorm(inplanes, λ=relu))
+    push!(layers, MaxPool((3, 3), stride=(2, 2), pad=(1, 1)))
+    inplanes = 64
+    outplanes = 64
+    for nrepeats in block_config
```

Comment:
We actually need to repeat the block `nrepeats` times. So the overall structure will look like:
```julia
for nrepeats in block_config
  for i in 1:nrepeats
    for expansion in channel_config
      # add a SkipConnection
    end
  end
  inplanes *= 2
end
```